app/src/main/java/org/mozilla/fenix/home/tabs/TabsComponent.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.tabs
+
+import android.view.ViewGroup
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import org.mozilla.fenix.mvi.Action
+import org.mozilla.fenix.mvi.ActionBusFactory
+import org.mozilla.fenix.mvi.Change
+import org.mozilla.fenix.mvi.UIComponent
+import org.mozilla.fenix.mvi.ViewState
+
+class TabsComponent(
+    private val sessionManager: SessionManager,
+    private val container: ViewGroup,
+    bus: ActionBusFactory,
+    override var initialState: TabsState = TabsState(sessionManager.sessions)
+) :
+    UIComponent<TabsState, TabsAction, TabsChange>(
+        bus.getManagedEmitter(TabsAction::class.java),
+        bus.getSafeManagedObservable(TabsChange::class.java)
+    ) {
+
+    override val reducer: (TabsState, TabsChange) -> TabsState = { state, change ->
+        when (change) {
+            is TabsChange.Changed -> state.copy(sessions = sessionManager.sessions)
```

Comment:
If we attach the sessionList to the `TabsChange.Changed` event we can create a new state from it. This will remove the requirement of the `SessionManager` and make this easier to re-use when we add private mode.

Reply:
While I still think this is a good change because it will make the reducer will be pure. It probably wont make a difference for private browsing 🙃